aiaccel/hpo/samplers/nelder_mead_sampler.py

```diff
@@ -0,0 +1,237 @@
+
+    def expand(self) -> Generator[np.ndarray[float, float], None, None]:
+        ye = self.yc + self.coef.e * (self.yc - self.vertices[-1])
+        self.storage.e = ye
+        yield from self.func(ye)
+
+    def inside_contract(self) -> Generator[np.ndarray[float, float], None, None]:
+        yic = self.yc + self.coef.ic * (self.yc - self.vertices[-1])
+        self.storage.ic = yic
+        yield from self.func(yic)
+
+    def outside_contract(self) -> Generator[np.ndarray[float, float], None, None]:
+        yoc = self.yc + self.coef.oc * (self.yc - self.vertices[-1])
+        self.storage.oc = yoc
+        yield from self.func(yoc)
+
+    def shrink(self) -> Generator[np.ndarray[float, float], None, None]:
+        for i in range(1, len(self.vertices)):
+            yield self.vertices[0] + self.coef.s * (self.vertices[i] - self.vertices[0]), 0
+        for i in range(1, len(self.vertices)):
+            vertex = self.vertex_queue.get()
+            self.vertices[i] = vertex.coordinate
+            self.values[i] = vertex.value
+
+    def func(self, y: np.ndarray[float, float]) -> Generator[np.ndarray[float, float], None, None]:
+        if self.is_within_range(y):
+            yield y, 0
+        else:
+            self.vertex_queue.put(Vertex(y, np.inf))
+
+    def is_within_range(self, coordinates: np.ndarray[float, float]) -> bool:
+        return all(not (co < ss[0] or ss[1] < co) for ss, co in zip(self._search_space.values(), coordinates))
+
+    def search(self) -> Generator[np.ndarray[float, float], None, None]:
+        yield from self.initial()
+        while True:
+
+            self.centroid()
+            yield from self.reflect()
+            fr = self.vertex_queue.get().value
+
+            if self.values[0] <= fr < self.values[-2]:
+
+                self.vertices[-1] = self.storage.r
+                self.values[-1] = fr
+
+            elif fr < self.values[0]:
+
+                yield from self.expand()
+                fe = self.vertex_queue.get().value
+
+                if fe < fr:
+                    self.vertices[-1] = self.storage.e
+                    self.values[-1] = fe
+
+                else:
+                    self.vertices[-1] = self.storage.r
+                    self.values[-1] = fr
+
+            elif self.values[-2] <= fr < self.values[-1]:
+
+                yield from self.outside_contract()
+                foc = self.vertex_queue.get().value
+
+                if foc <= fr:
+                    self.vertices[-1] = self.storage.oc
+                    self.values[-1] = foc
+
+                else:
+                    self.shrink()
+
+            elif self.values[-1] <= fr:
+
+                yield from self.inside_contract()
+                fic = self.vertex_queue.get().value
+
+                if fic < self.values[-1]:
+                    self.vertices[-1] = self.storage.ic
+                    self.values[-1] = fic
+
+                else:
+                    yield from self.shrink()
```

Comment:
shrinkが2回でてくるこの部分は，メソッドに分けず下記のように直せると良い気がしました．
```suggestion
            elif self.values[-2] <= fr < self.values[-1]:
                yield (oc := ...)
                foc = self.vertex_queue.get()

                if foc <= fr:
                    self.vertices[-1], self.values[-1] = oc, foc
                else:
                    shrink_requied = True
            elif self.values[-1] <= fr:
                yield (ic := ...)
                fic = self.vertex_queue.get()

                if fic < self.values[-1]:
                    self.vertices[-1], self.values[-1] = ic, fic
                else:
                    shrink_requied = True

            if is_shrink:
                pass  # shrinkの操作を書く
                shrink_requied = False
```

Reply:
対応しました。

71f29148f4b52151a458aecc2196d89b52d2d224

Reply:
こちら，
* yield (...) ficのあとの改行
* ic, ficの代入を1行に
* shrink_required = Falseは最後のif文のとこだけ
が反映頂けていないようなので修正お願いします．


他のverticesへの代入も三項演算子で1行にすると良いと思います (ruffで提案されると思います)．
また，Samplerの方にもcamel caseが残っているので，修正お願いします．

Reply:
反映しました。

93d182a24057f232756fd3e9b88416e4e03b26cc

>  (ruffで提案されると思います)

改修前に ruff を実行してみても、特に何も出てきませんでした。

```
$ ruff check --line-length 120 --fix aiaccel/hpo/samplers/nelder_mead_sampler.py
$ 
```

Reply:
ありがとうございます．1行の時だっただけかも…
この前確か提案された覚えがあるんですよね．

Reply:
```
shrink_required = False
```
がまだ残っていますので，反映をお願いします．

Reply:
あと，三項演算子の修正もお願いします…

Reply:
反映しました。

0dcfbe532a6b1ac96749f96cd3fb4a82c103e07d

Reply:
上記コミットの Optimizer Algorithm Test でエラーが発生しました。
`UnboundLocalError: local variable 'shrink_requied' referenced before assignment`

1度も shrink_requied を定義せずにif文の判定を行っていることが原因のようだったので、__iter__の最初に shrink_requied = False を追加しました。